tool/tctl/common/resource_command.go

```diff
@@ -419,6 +438,60 @@ func (rc *ResourceCommand) Delete(client auth.ClientI) (err error) {
 	return nil
 }
 
+// Update updates select resource fields: expiry and labels
+func (rc *ResourceCommand) Update(clt auth.ClientI) error {
+	if rc.ref.Kind == "" || rc.ref.Name == "" {
+		return trace.BadParameter("provide a full resource name to update, for example:\n$ tctl update rc/remote --set-labels\n")
+	}
+
+	var err error
+	var labels map[string]string
+	if rc.labels != "" {
+		labels, err = client.ParseLabelSpec(rc.labels)
+		if err != nil {
+			return err
+		}
+	}
+
+	var expiry *time.Time
+	if rc.ttl != "" {
+		duration, err := time.ParseDuration(rc.ttl)
+		if err != nil {
+			return err
+		}
+		exp := time.Now().UTC().Add(duration)
+		expiry = &exp
+	}
+
+	if expiry == nil && labels == nil {
+		return trace.BadParameter("use at least one of --set-labels or --set-ttl")
+	}
+
+	ctx := context.TODO()
+	switch rc.ref.Kind {
+	case services.KindRemoteCluster:
+		cluster, err := clt.GetRemoteCluster(rc.ref.Name)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if labels != nil {
+			meta := cluster.GetMetadata()
+			meta.Labels = labels
+			cluster.SetMetadata(meta)
```

Comment:
Nit: This block (get metadata, set labels, set metadata) has appeared at least 3 times in this PR making me think it might be worth providing `SetLabels()` shorthand on the resource interface.